nussl/datasets/hooks.py

```diff
@@ -417,6 +427,146 @@ def process_item(self, item):
+        verbose (bool, optional): suppress progress bar and logging for the
+         dataset preprocessing. Defaults to False.
+        **kwargs: Any additional arguments that are passed up to BaseDataset
+            (see ``nussl.datasets.BaseDataset``).
+    """
+
+    def __init__(self,
+                 folder: str,
+                 salient_src: str,
+                 sample_rate: int = None,
+                 mix_folder: str = 'mix',
+                 source_folders: List[str] = None,
+                 ext: List[str] = None,
+                 make_mix: bool = False,
+                 threshold_db: float = -60.0,
+                 segment_dur: float = 4.0,
+                 hop_ratio: float = 0.5,
+                 verbose: bool = False,
+                 balance: bool = False,
+                 **kwargs):
+        self.threshold_db = threshold_db
+        self.segment_dur = segment_dur
+        self.hop_ratio = hop_ratio
+        self.sample_rate = sample_rate
+        self.salient_src = salient_src
+        self.folder = folder
+        self.verbose = verbose
+        self.balance = balance
+
+        self.mix_src = MixSourceFolder(
+            folder, mix_folder, source_folders, ext, make_mix, **kwargs
+        )
+        self.song_metadata = self._populate_metadata()
+        self.offset = 0.0
+
+        super(SalientExcerptMixSourceFolder, self).__init__(
+            self._get_mix,
+            len(self.song_metadata),
+            **kwargs
+        )
+
+    def _populate_metadata(self):
+        """Makes a metadata dict of salient start times for each source."""
+
+        if self.verbose:
+            logging.info(f'Preprocessing data at {self.folder}.')
+
+        items = self.mix_src.get_items(self.folder)
+        balanced_samples = []
+        metadata = []
+        for item in tqdm.tqdm(items, disable=not self.verbose):
+            mix, sources = self.mix_src.get_mix_and_sources(item)
+            target_src = sources[self.salient_src]
+            target_audio = target_src.audio_data
+            salient_starts = utils.find_salient_starts(target_audio,
+                                                       self.segment_dur,
+                                                       self.hop_ratio,
+                                                       mix.sample_rate,
+                                                       self.threshold_db)
+            # this is a fairly cheap operation, no need to embed it within a
+            # conditional
+            balanced_samples.append((len(salient_starts), item,
+                                     salient_starts, mix.sample_rate))
+            for start in salient_starts:
+                if (start + self.segment_dur * mix.sample_rate) <= len(mix):
+                    metadata.append({
+                        'mixsrc_item': item,
+                        'start': start / mix.sample_rate
+                    })
+        # if the balance flag is set then balance the metadata
```

Comment:
Same with this comment. In general, use inline comments to explain code excerpts that might be difficult to parse like complex logic, a seemingly unorthodox decision, or the expected contents of a data structure. Adding unnecessary comments can clutter up the code.

A more effective comment here would be like `# self._balance_set() will take care of the balancing logic`. But I'm still not convinced you need a comment here though...